roles/cifmw_cephadm/tasks/rgw_cert.yml

```diff
@@ -0,0 +1,119 @@
+---
+# Copyright 2024 Red Hat, Inc.
+# All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License. You may obtain
+# a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+
+- name: Check if cifmw_cephadm_certificate path exists
+  ansible.builtin.stat:
+    path: "{{ cifmw_cephadm_certificate }}"
+  register: cert_stat_result
+  when: cifmw_cephadm_certificate | length > 0
+
+- name: Check if cifmw_cephadm_key path exists
+  ansible.builtin.stat:
+    path: "{{ cifmw_cephadm_key }}"
+  register: key_stat_result
+  when: cifmw_cephadm_key | length > 0
+
+- name: Create certificate if paths do not exist on target host
+  when: (not cert_stat_result.stat.exists | bool) or
+        (not key_stat_result.stat.exists | bool)
+  block:
+    - name: Ensure we have cifmw_openshift_kubeconfig
+      delegate_to: localhost
+      ansible.builtin.stat:
+        path: "{{ cifmw_openshift_kubeconfig }}"
+      register: cifmw_openshift_kubeconfig_stat_result
+
+    - name: Fail if kube config file is missing
+      delegate_to: localhost
+      fail:
+        msg: >-
+          Missing kube config file ({{ cifmw_openshift_kubeconfig }})
+          Canot create RGW certificate without access to root CA in k8s
+      when:
+        - not cifmw_openshift_kubeconfig_stat_result.stat.exists | bool
+
+    - name: Create Ceph certificate manifest directory
+      delegate_to: localhost
+      ansible.builtin.file:
+        path: "{{ cifmw_cephadm_manifests_dir }}"
+        state: directory
+
+    - name: Put the certificate manifest file(s) in place
+      delegate_to: localhost
+      ansible.builtin.template:
+        src: "templates/{{ item }}.yml.j2"
+        dest: "{{ cifmw_cephadm_manifests_dir }}/{{ item }}.yml"
+        mode: "0644"
+        force: true
+      loop:
+        - cert_ceph_rgw
```

Comment:
this is interesting: in theory we can make this generic enough to not only generate the cert for rgw, but do the same thing for the Ceph dashboard. However I understand this is out of scope and we might want to follow up for that scenario.
I see that the next task has `cert_ceph_rgw` hardcoded, so we can choose to either remove the loop here, or add the loop to the next task as well.

Reply:
Also, I'm wondering if we can use `cifmw_cephadm_cert_name` here

Reply:
I can use `cifmw_cephadm_cert_name` here for the `dest`.

However, `roles/cifmw_cephadm/templates/cert_ceph_rgw.yml.j2` has to be hard coded for the `src` since it's an actual template file which has a specific name. However, I name that file in the templates direcotry needs to be referenced by name in the `src` else I'll get something like:

```
    Could not find or access 'templates/cert-ceph-rgw.yml.j2'
```